fiona/_err.pyx

```diff
@@ -226,6 +226,18 @@ cdef inline object exc_check():
         return
 
 
+cdef int exc_wrap_int(int err) except -1:
```

Comment:
I tried writing something like this in an earlier PR. I think there might be some functions which return a non-zero error code, but don't set the GDAL exception. I can't remember the specifics though. I think it's worth adding an `else` to this function like `exc_wrap_pointer` raises `FionaNullPointerError`. It may never happen, but if it does and we skip over a non-zero error it can be difficult to debug.